### DISCUSSION TITLE
Window padding

### DISCUSSION THREAD

**Discussion by alejoguitar** [contour-terminal/contour#1226] [Q&A]
Is there a way to set some padding on the window?
I'm using Contour for the first time, I think I read all the available documentation, but I can't find a way to do it.

Here's a Contour window running the lf file manager, it would be great if I could separate the first line of text from the window border.

![Captura desde 2023-09-30 23-28-27](https://github.com/contour-terminal/contour/assets/100370632/23d451dd-e6aa-481d-b91e-57f22867db36)

**Comment by Yaraslaut** [comment]
If you want to create round corners for the window you need to add borders to that window as well. See 
![image](https://github.com/contour-terminal/contour/assets/44506630/6e1a5524-0ff2-4fca-be12-1c8daf3ee96d)
So the corner will not overlap with the actual window 
In my case i have to add border to the window from window manager (awesomwm) and then set radius of the border inside composer (picom)

**Comment by christianparpart** [comment]
Hi @alejoguitar, thanks for trying out Contour terminal.

With respect to padding, the terminal's viewport definitely should support it, but we sadly did not implement it just yet. See #126 for the feature request.

@Yaraslaut maybe we should prioritize that slightly and make it part of the next release (0.4.0), too? What do you think? It should be not too complex to implement. What made me afraid of doing is, is getting the calculations wrong. Mouse cursor position to grid cell mapping must be adapted then too.

**Reply by christianparpart** [reply]
p.s.: the problem might be workaroundable on awesomewm, but we still have that issue on MacOS, where we can't control that :)

**Reply by Yaraslaut** [reply]
Hm, maybe we can implement this features together with tabs, since we will need another positioning of the window at that point anyway
